File: Principal/AppDrive.cs
using DigoFramework;
using Drive.Server;
using Drive.Server.WebSocket;
using NetZ.Web;
using NetZ.Web.Server;
using System.Collections.Generic;
using System.IO;

namespace Drive
{
    internal class AppDrive : AppWebBase
    {
        #region Constantes

        public const long INT_TRANSFERENCIA_PARTE_TAMANHO = 1000000;

        #endregion Constantes

        #region Atributos

        private static AppDrive _i;

        public new static AppDrive i
        {
            get
            {
                if (_i != null)
                {
                    return _i;
                }

                _i = new AppDrive();

                return _i;
            }
        }

        #endregion Atributos

        #region Construtores

        private AppDrive() : base("Drive")
        {
        }

        #endregion Construtores

        #region Métodos

        protected override ConfigWebBase getObjConfig()
        {
            return ConfigDrive.i;
        }

        protected override TemaBase getObjTema()
        {
            return new TemaDrive();
        }

        protected override void inicializar()
        {
            base.inicializar();

            if (string.IsNullOrEmpty(ConfigDrive.i.dirRepositorio))
            {
                new Erro("O repositório de arquivos não foi indicado.");
            }
            else
            {
                Directory.CreateDirectory(ConfigDrive.i.dirRepositorio);
            }
        }

        protected override void inicializarLstSrv(List<ServerBase> lstSrv)
        {
            lstSrv.Add(new SrvHttpDrive());

            lstSrv.Add(SrvWsDrive.i);
        }

        #endregion Métodos

        #region Eventos

        #endregion Eventos
    }
}

File: Drive.TypeScript/AppDrive.ts
/// <reference path="../Web.TypeScript/AppWebBase.ts"/>
/// <reference path="../Web.TypeScript/server/ServerBase.ts"/>

module Drive
{
    // #region Importações

    import AppWebBase = Web.AppWebBase;
    import ServerBase = Web.ServerBase;

    // #endregion Importações

    // #region Enumerados
    // #endregion Enumerados

    export class AppDrive extends AppWebBase
    {
        // #region Constantes
        // #endregion Constantes

        // #region Atributos

        protected static _i: AppDrive;

        public static get i(): AppDrive
        {
            if (AppDrive._i != null)
            {
                return AppDrive._i;
            }

            AppDrive._i = new AppDrive();

            return AppDrive._i;
        }

        // #endregion Atributos

        // #region Construtores
        // #endregion Construtores

        // #region Métodos

        protected inicializarArrSrv(arrSrv: Array<ServerBase>): void
        {
            super.inicializarArrSrv(arrSrv);

            arrSrv.push(SrvWsDrive.i);
        }

        // #endregion Métodos

        // #region Eventos
        // #endregion Eventos
    }

    // #region Inicialização

    $(document).ready(() => { AppDrive.i.iniciar(); });

    // #endregion Inicialização
}